### ISSUE TITLE
insider: git-committers shows GitHub, even for GitLab repositories

### ISSUE THREAD

**Issue by ojacques** [squidfunk/mkdocs-material#6283]
### Bug description

When using the [mkdocs-git-committers-plugin-2](https://github.com/ojacques/mkdocs-git-committers-plugin-2) v2.1.0, which now supports [GitLab repositories](https://github.com/ojacques/mkdocs-git-committers-plugin-2/releases/tag/2.1.0), the contributors are next to a GitHub logo and text.

![image](https://github.com/squidfunk/mkdocs-material/assets/5385290/cdd73151-ed2d-427f-bb01-dbb62ba02250)

I expect this to be GitLab logo and text.

### Related links

- [[Reporting a bug](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/)
-](https://squidfunk.github.io/mkdocs-material/setup/adding-a-git-repository/#document-contributors)
- https://github.com/ojacques/mkdocs-git-committers-plugin-2/issues/28#issuecomment-1791713257

### Reproduction

[demo.zip](https://github.com/squidfunk/mkdocs-material/files/13245726/demo.zip)

### Steps to reproduce

1. Start a new mkdocs material site on gitlab
2. Activate the committers plugin:
```
plugins:
  - git-committers:
      gitlab_repository: 12345678
      token: !ENV ["GH_TOKEN"]
```
3. Render the site
4. At the bottom of the page, see GitHub logo and text

![image](https://github.com/squidfunk/mkdocs-material/assets/5385290/2cbd43cb-882d-4bdb-b0a4-f756e61ec7ff)

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material/), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/).

**Comment by squidfunk** [comment]
Thanks for reporting. How can we distinguish this in the template? We need some way to decide what icon to show with a simple `{% if %}...{% else %}` condition. [See the relevant section here](https://github.com/squidfunk/mkdocs-material-insiders/blob/e7a6a238e163b86b435dcb77d5b706b3e453ab7a/src/templates/partials/source-file.html#L83-L90) (Insiders).

**Comment by ojacques** [comment]
I was thinking that checking `author.url` for github or gitlab would work. But for self-hosted GitHub or GitLab, you can't be sure that the url will contain github or gitlab.
Is that OK for you if I add a `source` field (`author.source`) to the author object, and make it either `github` or `gitlab`?

**Comment by squidfunk** [comment]
> Is that OK for you if I add a source field (author.source) to the author object, and make it either github or gitlab?

Sounds perfect! Once that's implemented, we can directly make the necessary changes – it's just changing an icon after all.

**Comment by ojacques** [comment]
Done in [mkdocs-git-committers-plugin-2 2.2.0](https://github.com/ojacques/mkdocs-git-committers-plugin-2/releases/tag/2.2.0).

**Comment by squidfunk** [comment]
Thanks! I'm not sure how to use it. Is `source` defined per author? That's kind of weird, because we would need to hoist the value then, and it is the same for all authors, isn't it? Our templates are like this:

```
SOURCE
each AUTHOR of AUTHORS
  AUTHOR
```

**Comment by ojacques** [comment]
Hmmm. I thought it was easier for you to have source repeated for all authors (although, I agree it does not make sense, it is the same for all authors of a given repo). 
I will hoist source up in the context, in `committers-source` (I don't want to override another potential `source`).

**Comment by ojacques** [comment]
Done in https://github.com/ojacques/mkdocs-git-committers-plugin-2/releases/tag/2.2.1.
`committers-source` equals `github` or `gitlab`.

**Comment by squidfunk** [comment]
Sorry to bother again, but as far as I know, a variable cannot include `-` in its name. At least I don't know how to reference them in templates. I would recommend switching to `_`. I've never seen variables using `-` 😉 Error:

```
...
    {% if committers-source == "gitlab" %}
^^^^^^^^^^^^^^^^^^^^^^^^^
jinja2.exceptions.UndefinedError: 'source' is undefined
```

**Comment by squidfunk** [comment]
I've made all changes, took the opportunity to modularize the template (to make it ready for more integrations with git-related plugins) and pushed everything in [`3b7632a`](https://github.com/squidfunk/mkdocs-material-insiders/commit/3b7632a9f7d95898b2517f0273835a6f71e191ac). If you switch `committers-source` to `committers_source` it should work right away. The default is that it falls back to GitHub, so it will also work with older versions of the plugin 🚀

**Comment by ojacques** [comment]
All set in [2.2.2](https://github.com/ojacques/mkdocs-git-committers-plugin-2/releases/tag/2.2.2). 
Thanks!

**Comment by squidfunk** [comment]
Released as part of [9.4.8+insiders-4.43.0](https://squidfunk.github.io/mkdocs-material/insiders/changelog/#4.43.0).
